detection_rules/config.py

```diff
@@ -145,6 +145,7 @@ class RulesConfig:
     test_config: TestConfig
     version_lock_file: Path
     version_lock: Dict[str, dict]
+    version_strategy: Literal['auto', 'manual'] = "auto"
```

Comment:
```suggestion
    ignore_version_lock: bool = False
```

or bypass. Should be opt out explicitly